pkg/controller/rhmiconfig/helpers/update_status.go

```diff
@@ -0,0 +1,101 @@
+package helpers
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const WINDOW = 6
+
+func UpdateStatus(ctx context.Context, client k8sclient.Client, config *integreatlyv1alpha1.RHMIConfig, targetVersion string) error {
```

Comment:
```suggestion
func UpdateStatus(config *integreatlyv1alpha1.RHMIConfig) error {
```

I'd suggest to make this function simply update the status of the RHMIConfig in memory, as `client.Status().Update()` is already being called afterwards, to avoid calling it twice unnecessarily. We can get rid of the context and client parameters then.

You can safely get rid of the `targetVersion` parameter. If you notice you're setting the `UpgradeAvailable.TargetVersion` field with this value, but then passing `UpgradeAvailable.TargetVersion` as the parameter, effectively doing nothing.

Reply:
Thanks, @sergioifg94, your suggested changes around UpdateStatus make a lot of sense!